store/sqlstore/channel_store.go

```diff
@@ -1936,3 +1936,57 @@ func (s SqlChannelStore) ResetLastPostAt() store.StoreChannel {
 		}
 	})
 }
+
+func (s SqlChannelStore) GetAllChannelsForExportAfter(limit int, afterId string) store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
+		var data []*model.ChannelForExport
+		if _, err := s.GetReplica().Select(&data, `
+			SELECT
+				Channels.*,
+				Teams.Name as TeamName,
+				Schemes.Name as SchemeName
+			FROM Channels
+			INNER JOIN
+				Teams ON Channels.TeamId = Teams.Id
+			LEFT JOIN
+				Schemes ON Channels.SchemeId = Schemes.Id
+			WHERE
+				Channels.Id > :AfterId
+				AND Channels.Type IN ('O', 'P')
```

Comment:
I miss here the export of direct channels and group channels, I don't know if is on purpose.

Reply:
Yeah, it's intentional. The data structure for them is different, so they need to come out in a different query. I haven't implemented that in this PR as the idea was to be as minimal as possible - we can add them later.